### DISCUSSION TITLE
Use rich library to reduce codebase

### DISCUSSION THREAD

**Discussion by FlorianWilhelm** [pyscaffold/pyscaffold#344] [Ideas]
Since in PyScaffold 4.0 external dependencies will be no problem (in contrast to how it was before), would it make sense to use [rich](https://github.com/willmcgugan/rich) in order to decrease the codebase of PyScaffold? For instance, parts of `termui.py` and `log.py`?

**Comment by abravalheri** [comment]
This is a very cool library. I think it definetely makes sense to try to use it.

**Comment by julie777** [comment]
I still don't know enough to know where or why pyscaffold is colorizing things.
Maybe someone could point out where this happens and why it is important?

**Reply by FlorianWilhelm** [reply]
Colorizing is done in when logging, i.e. with the `--pretend`, `--list-actions`, `--verbose` and `--very-verbose` flags. It was merely done to help users understanding faster what is going on by color grouping similar actions. I think it's not extremely necessary but still a very helpful feature. A lot of CLI tools do this nowadays and if colours are used for a reason, like structuring the text output, I think it's valid.

This issue is quite an old one and one would really need to see if it helps us in reducing the codebase and maybe its complexity. We want to develop and maintain PyScaffold for many, many years to come and also adapt it to whatever the future of Python packing holds for us. Thus, we need the codebase to be clean, simple and reduced to the bar minimum we need. It would be interesting to get your feedback on this as you are now trying to understand it. Be honest with us ;-) I actually find our code right now already quite complex and would love to see some simplifications in the future but that's also a bit of personal taste.

**Reply by julie777** [reply]
interesting. I wonder why I am not seeing any colorizing.

I agree with you about complexity. On one hand my initial impression was that there were some nice architectural concepts such as actions, but I definitely noticed some things that seemed like they could be simplified.
